### PR TITLE
Collection support

### PR DESCRIPTION
**Pull request by gundalow**
Copy of `gundalow:collections` branch https://github.com/gundalow/ansibullbot/tree/collections

### CODE REVIEW

File: ansibullbot/utils/gh_gql_client.py [hunk 1]
```diff
@@ -377,5 +377,5 @@ def requests(self, payload):
     logging.basicConfig(level=logging.DEBUG)
     client = GithubGraphQLClient(C.DEFAULT_GITHUB_TOKEN)
     summaries = client.get_all_summaries(u'ansible', u'ansible')
-    ln = client.get_last_number(u'ansible/ansible')
+    ln = client.get_last_number(u'ansible-collections/community.general')
```
Comment (gundalow) [inline ansibullbot/utils/gh_gql_client.py:380]: This looks wrong

File: ansibullbot/triagers/ansible.py [hunk 2]
```diff
@@ -834,7 +864,8 @@ def create_actions(self, iw, actions):
         if not self.meta[u'is_bad_pr']:
             if iw.is_issue() and self.meta.get(u'needs_component_message'):
                 tvars = {
-                    u'meta': self.meta
+                    u'meta': self.meta,
+                    u'base_url': 'https://github.com/%s/blob/%s/' % (self.collection,repo_data[self.collection][u'branch']),
```
Comment (gundalow) [inline ansibullbot/triagers/ansible.py:868]: @dmsimard Will need this change upstream

File: ansibullbot/triagers/ansible.py [hunk 3]
```diff
@@ -1049,9 +1080,13 @@ def create_actions(self, iw, actions):
                     actions.unlabel.append(u'needs_ci')
 
         # MODULE CATEGORY LABELS
+        # {u'cloud/amazon': u'aws', u'cloud/azure': u'azure', u'network': u'networking', u'cloud/google': u'gce', u'windows': u'windows', u'cloud/openstack': u'openstack', u'cloud/digital_ocean': u'digital_ocean', u'cloud': u'cloud'}
+        # FIXME This will want testing once we have subdirectories under plugins/modules/
         if not self.meta[u'is_bad_pr']:
-            if self.meta[u'is_new_module'] or self.meta[u'is_module']:
+            # I think we want labels on everything
+            if True: # self.meta[u'is_new_module'] or self.meta[u'is_module']:
```
Comment (gundalow) [inline ansibullbot/triagers/ansible.py:1087]: @dmsimard Will need this change upstream

File: ansibullbot/triagers/ansible.py [hunk 4]
```diff
@@ -1411,24 +1437,6 @@ def create_actions(self, iw, actions):
                         actions.unlabel.append(u'needs_maintainer')
 
         # https://github.com/ansible/ansibullbot/issues/608
-        if not self.meta[u'is_bad_pr']:
```
Comment (gundalow) [inline ansibullbot/triagers/ansible.py:1414]: I'm not sure why I removed this

File: ansibullbot/triagers/ansible.py [hunk 5]
```diff
@@ -2002,13 +2015,13 @@ def process(self, iw):
         self.meta.update(
             get_shipit_facts(
                 iw, self.meta, self.module_indexer,
-                core_team=self.ansible_core_team, botnames=self.BOTNAMES
```
Comment (gundalow) [inline ansibullbot/triagers/ansible.py:2005]: Might need this

File: ansibullbot/utils/component_tools.py [hunk 6]
```diff
@@ -179,32 +178,23 @@ def index_files(self):
...
-            if len(parts) != 2 or checkoutdir not in line:
-                continue
-
-            mname = parts[0]
-            if mname not in self.MODULE_NAMES:
-                self.MODULE_NAMES.append(mname)
-
-            fpath = parts[1]
-            fpath = fpath.replace(checkoutdir + u'/', u'')
+        for root, directories, filenames in os.walk(os.path.join(checkoutdir, 'plugins', 'modules')):
```
Comment (gundalow) [inline ansibullbot/utils/component_tools.py:181]: Not sure if we need this change still.

**Review by gundalow — COMMENTED** [review]
Changes that I think we still need to make to wip_upstream (or directly to upstream)

File: ansibullbot/utils/component_tools.py (hunk 6 above)
Comment (dmsimard) [reply ansibullbot/utils/component_tools.py:181]: I believe this has been moved out to https://github.com/ansible/ansibullbot/blob/d3274fb73488cfc3a7b1d9321360ea9153fafb96/ansibullbot/utils/git_tools.py#L64-L66

It's no longer here: https://github.com/ansible/ansibullbot/blob/master/ansibullbot/utils/component_tools.py#L293-L302

**Pull request merged**